lib/graphql.ex

```diff
@@ -22,7 +25,30 @@ defmodule GraphQL do
   def execute(schema, query, root_value \\ %{}, variable_values \\ %{}, operation_name \\ nil) do
     case GraphQL.Lang.Parser.parse(query) do
       {:ok, document} ->
-        case GraphQL.Execution.Executor.execute(schema, document, root_value, variable_values, operation_name) do
+        case Validator.validate(schema, document) do
```

Comment:
I'd like to do that but then I tried to write it without duplication and
made it worse. I'll try again.

On Monday, 28 March 2016, Josh Price <notifications@github.com> wrote:

> In lib/graphql.ex
> <https://github.com/graphql-elixir/graphql/pull/79#discussion_r57551261>:
>
> > @@ -22,7 +25,30 @@ defmodule GraphQL do
> >    def execute(schema, query, root_value \\ %{}, variable_values \\ %{}, operation_name \\ nil) do
> >      case GraphQL.Lang.Parser.parse(query) do
> >        {:ok, document} ->
> > -        case GraphQL.Execution.Executor.execute(schema, document, root_value, variable_values, operation_name) do
> > +        case Validator.validate(schema, document) do
>
> Can we remove the duplication here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/graphql-elixir/graphql/pull/79/files/87e212e5289def1e38168ffc4e6c2ea1ddf730de#r57551261>
>


-- 
James